### DISCUSSION TITLE
[Bug]: `showPanel: false` not working (`v8.0.0-beta.0`)

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [storybookjs/storybook#25913] [Help] [answered]
### Describe the bug

`.storybook/manager.ts`

```ts
import { addons } from "@storybook/manager-api";
import { create } from "@storybook/theming/create";

addons.setConfig({
  showPanel: false,
  theme: create({
    base: "dark",
  }),
});
```

After upgrading to `v8.0.0-beta.0`, the addons panel appears when I start Storybook, even though I've turned it off.

I still get a dark theme, which indicates `manager.ts` is still being read and understood; it's just the `showPanel: false` setting that isn't being respected.

### To Reproduce

https://stackblitz.com/edit/github-izp6au-fo1t1m?file=.storybook%2Fmanager.ts

StackBlitz takes about a year to run Storybook, but the issue does reproduce. You'll have to open up the right pane wide enough to confirm.

<img width="1386" alt="screenshot of bug" src="https://github.com/storybookjs/storybook/assets/1505561/8634e2d2-1c47-431a-af58-b375c22bd440">

**Comment by vanessayuenn** [comment, marked as answer]
@aaronadamsCA thanks for trying out Storybook 8 beta! Indeed, starting from Storybook 8, `showPanel` is no longer a valid option. It has instead been split out to `bottomPanelHeight: number` and `rightPanelWidth: number`, where the numbers represents the size of the panel in pixels. Here is the [relevant migration notes section](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#ui-layout-state-has-changed-shape), and the [API docs](https://storybook.js.org/docs/8.0/configure/features-and-behavior).

**Reply by aaronadamsCA** [reply]
I humbly submit that `MIGRATION.md` has grown way, way too long! Missing that section is completely on me—except I didn't even realize I'd missed it.

**Comment by gossi** [comment]
I followed the bug report to this discussion and the new API is not working for me.

I set `bottomPanelHeight: 350` for one story and `bottomPanelHeight: 0` for another in the same file. I cannot say when storybook will give the panel some height or not, but it would show the panel for both stories or it would hide it for both, but wouldn't switch between.

Second, the API is somewhat inconsistent, no? I can set:

```ts
export default {
  // ...
  parameters: {
    options: {
      rightPanelWidth: 200,
      bottomPanelHeight: 350
    }
  }
};
```

"Oh yes, please do both!" - I _assume_ some field takes precedence but that is not visible to the user as per API.

A more prescriptive API is this:

```ts
export default {
  // ...
  parameters: {
    options: {
      panelSize: 200, // similar to flex-basis
      panelPosition: 'right'
    }
  }
};
```

I would make it:

```ts
panelSize: number | boolean;
```

- `number` - giving an explicit size over the control 
- `boolean` - `true` - show it, to whatever size it was before and `false` to hide it

And for most stories I've written I actually didn't care that much about the size. This API would benefit both use-cases, while giving guarantees about what is actually happening.

**Reply by aaronadamsCA** [reply]
I will add my +1 that the new "improved" API for this is very clearly worse than before!

**Reply by eliasmohsengrenvald** [reply]
I also tried the `rightPanelWidth` and `bottomPanelHeight` options and can also confirm that it would either show the panel for all stories or it would hide it for all!
